### PR TITLE
Add documentation regarding specification URL and OData

### PR DESCRIPTION
**Pull request by Szymongib**
**Description**

Changes proposed in this pull request:

- Document supported API types
- Document `SpecificationUrl`

**Related issue(s)**
#2935

### CODE REVIEW

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 1]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
     ```
     curl https://gateway.kyma.local:{NODE_PORT}/{APP_NAME}/v1/metadata/services/{YOUR_SERVICE_ID} --cert {CERT_FILE_NAME}.crt --key {KEY_FILE_NAME}.key -k
     ```
+
+
+## Register API with specification URL
+
+Application Registry allows you to pass API specification in form of specification URL.
+
+To register API with specification URL replace `api.spec` with `api.specificationUrl`. Be aware that `api.spec` has higher priority than `api.specificationUrl`.
```
Comment (akgalwas) [inline docs/application-connector/docs/08-03-register-manage-services.md]: I would rephrase this sentence to state that if you provide both api.spec and api.specificationUrl then api.spec will be used.

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 2]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
...
+  "specificationUrl": "https://services.odata.org/OData/OData.svc/$metadata",
+  "credentials": {
+    "basic": {
+      "username": "{USERNAME}",
+      "password": "{PASSWORD}"
+    }
+}
+```
+
+The Application Registry will fetch the specification from provided URL but it will not use any credentials, therefor the endpoint can not be secured by any authentication mechanism.
```
Comment (akgalwas) [inline docs/application-connector/docs/08-03-register-manage-services.md]: Typo "therefor"

File: docs/application-connector/docs/03-07-api-registration.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+title: API registration in Application Registry
+type: Details
+---
+
+### Supported formats
+
+The Application Registry supports API Specification in the following formats:
+- OpenAPI 2.0
+- OData 2.0
```
Comment (akgalwas) [inline docs/application-connector/docs/03-07-api-registration.md]: OData XML is supported in the following versions: 2.0,3.0,4.0. OData JSON is not supported by documentation subsystem and won't be displayed in console.

File: docs/application-connector/docs/03-07-api-registration.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
+title: API registration in Application Registry
+type: Details
+---
+
+### Supported formats
+
+The Application Registry supports API Specification in the following formats:
```
Comment (klaudiagrz) [inline docs/application-connector/docs/03-07-api-registration.md]:
```suggestion
The Application Registry supports the following formats of the API specification:
```

File: docs/application-connector/docs/03-07-api-registration.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+- OpenAPI 2.0
+- OData 2.0
+
+### Registration methods
+
+Application Registry supports two ways of passing API spec:
```
Comment (klaudiagrz) [inline docs/application-connector/docs/03-07-api-registration.md]:
```suggestion
You can pass the API specification in two ways:
```

File: docs/application-connector/docs/03-07-api-registration.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+- in JSON format
```
Comment (klaudiagrz) [inline docs/application-connector/docs/03-07-api-registration.md]:
```suggestion
- JSON format
```

File: docs/application-connector/docs/03-07-api-registration.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+- by `SpecificationUrl`
```
Comment (klaudiagrz) [inline docs/application-connector/docs/03-07-api-registration.md]:
```suggestion
- `SpecificationUrl`
```

File: docs/application-connector/docs/03-07-api-registration.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+>**NOTE:** Specification passed directly as a JSON has a higher priority than `SpecificationUrl`, which in such case will be ignored.
+
+For OpenAPI format both methods are supported.
```
Comment (klaudiagrz) [inline docs/application-connector/docs/03-07-api-registration.md]:
```suggestion
For the OpenAPI format, both methods are supported.
```

File: docs/application-connector/docs/03-07-api-registration.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+OData APIs can only be registered with `SpecificationUrl`.
```
Comment (klaudiagrz) [inline docs/application-connector/docs/03-07-api-registration.md]:
```suggestion
You can register OData APIs only with `SpecificationUrl`.
```

File: docs/application-connector/docs/03-07-api-registration.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+- OpenAPI 2.0
+- OData 2.0
+
+### Registration methods
+
+Application Registry supports two ways of passing API spec:
+- in JSON format
+- by `SpecificationUrl`
+
+>**NOTE:** Specification passed directly as a JSON has a higher priority than `SpecificationUrl`, which in such case will be ignored.
```
Comment (klaudiagrz) [inline docs/application-connector/docs/03-07-api-registration.md]:
```suggestion
>**NOTE:** Specification passed directly as a JSON has a higher priority than `SpecificationUrl`.  If you use these two methods at once, `SpecificationUrl` is ignored.
```

Comment (klaudiagrz) [reply docs/application-connector/docs/03-07-api-registration.md]: I'd move this note at the end of the document.

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 11]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
     ```
     curl https://gateway.kyma.local:{NODE_PORT}/{APP_NAME}/v1/metadata/services/{YOUR_SERVICE_ID} --cert {CERT_FILE_NAME}.crt --key {KEY_FILE_NAME}.key -k
     ```
+
+
+## Register API with specification URL
+
+Application Registry allows you to pass API specification in form of specification URL.
```
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]:
```suggestion
The Application Registry allows you to pass API specifications in a form of specification URLs.
```

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 12]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
     ```
     curl https://gateway.kyma.local:{NODE_PORT}/{APP_NAME}/v1/metadata/services/{YOUR_SERVICE_ID} --cert {CERT_FILE_NAME}.crt --key {KEY_FILE_NAME}.key -k
     ```
+
+
+## Register API with specification URL
```
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]:
```suggestion
## Register API with a specification URL
```

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 1 above)
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]:
```suggestion
To register API with a specification URL, replace **api.spec** with **api.specificationUrl**. Note that **api.spec** has a higher priority than **api.specificationUrl**.
```

Comment (klaudiagrz) [reply docs/application-connector/docs/08-03-register-manage-services.md]: ``Be aware that `api.spec` has higher priority than `api.specificationUrl`.`` -> so what? Please provide more info.

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 13]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
...
+
+Application Registry allows you to pass API specification in form of specification URL.
+
+To register API with specification URL replace `api.spec` with `api.specificationUrl`. Be aware that `api.spec` has higher priority than `api.specificationUrl`.
+
+```
```
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]: What is that? Please describe the example in one sentence, for example:
`See the example of blablabla:`

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 2 above)
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]:
```suggestion
The Application Registry fetches specification from the provided URL but it does not use any credentials, therefore the endpoint cannot be secured by any authentication mechanism.
```

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 14]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
...
+    "basic": {
+      "username": "{USERNAME}",
+      "password": "{PASSWORD}"
+    }
+}
+```
+
+The Application Registry will fetch the specification from provided URL but it will not use any credentials, therefor the endpoint can not be secured by any authentication mechanism.
+
+>**NOTE:** Fetching specification from URL is supported only for API Spec. It can not be done for Events and Documentation.
```
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]:
```suggestion
>**NOTE:** Fetching specification from a URL is supported only for APIs. Fetching specifications for Events or documentation is not supported.
```

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 15]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
...
+
+
+## Registering OData API
```
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]:
```suggestion
## Register the OData API
```

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 16]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
...
+
+If no `api.spec` or `api.specificationUrl` are specified and `api.type` is set to `OData`, the Application Registry will try to fetch the specification from the target URL with the `$metadata` path.
```
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]:
```suggestion
If the **api.spec** or **api.specificationUrl** parameters are not specified and the **api.type** parameter is set to `OData`, the Application Registry will try to fetch specification from the target URL with the `$metadata` path.
```

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 17]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
...
+
+For service with the following api:
```
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]:
```suggestion
For example, for the service with the following API, the Application Registry will try to fetch API specification from `https://services.odata.org/OData/OData.svc/$metadata`.
```

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 18]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
...
+  "apiType": "OData"
+  "credentials": {
+    "basic": {
+      "username": "{USERNAME}",
+      "password": "{PASSWORD}"
+    }
+}
+```
+
+Application Registry will try to fetch API Spec from `https://services.odata.org/OData/OData.svc/$metadata`.
```
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]: Delete.

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 1 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: Done!

File: docs/application-connector/docs/03-07-api-registration.md (hunk 3 above)
Comment (franpog859) [reply docs/application-connector/docs/03-07-api-registration.md]: Done!

File: docs/application-connector/docs/03-07-api-registration.md [hunk 19]
```diff
@@ -0,0 +1,21 @@
+---
+title: API registration in Application Registry
```
[the remark this answers is not in the document]
Comment (franpog859) [reply docs/application-connector/docs/03-07-api-registration.md]: Done!

File: docs/application-connector/docs/03-07-api-registration.md [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+type: Details
+---
+
+### Supported formats
```
[the remark this answers is not in the document]
Comment (franpog859) [reply docs/application-connector/docs/03-07-api-registration.md]: Done!

File: docs/application-connector/docs/03-07-api-registration.md (hunk 4 above)
Comment (franpog859) [reply docs/application-connector/docs/03-07-api-registration.md]: Done!

File: docs/application-connector/docs/03-07-api-registration.md (hunk 5 above)
Comment (franpog859) [reply docs/application-connector/docs/03-07-api-registration.md]: Done!

File: docs/application-connector/docs/03-07-api-registration.md (hunk 7 above)
Comment (franpog859) [reply docs/application-connector/docs/03-07-api-registration.md]: Done!

File: docs/application-connector/docs/03-07-api-registration.md (hunk 8 above)
Comment (franpog859) [reply docs/application-connector/docs/03-07-api-registration.md]: Done!

File: docs/application-connector/docs/03-07-api-registration.md (hunk 9 above)
Comment (franpog859) [reply docs/application-connector/docs/03-07-api-registration.md]: Done!

File: docs/application-connector/docs/03-07-api-registration.md (hunk 10 above)
Comment (franpog859) [reply docs/application-connector/docs/03-07-api-registration.md]: Done!

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 11 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: Done!

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 12 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: Done!

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 1 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: Done!

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 2 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: Done!

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 14 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: Done!

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 15 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: Done!

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 16 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: Done!

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 17 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: Done!

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 18 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: Done!

File: docs/application-connector/docs/08-03-register-manage-services.md (hunk 13 above)
Comment (franpog859) [reply docs/application-connector/docs/08-03-register-manage-services.md]: That's the API part of the request body if the user wants to register their API with specification URL. 
```
See the example of the API part of the request body with specification URL:
...
>**NOTE** If both api.spec and api.specificationUrl are provided, api.spec will be used due to its higher priority.
```
Would that be better, @klaudiagrz ?

Comment (klaudiagrz) [reply docs/application-connector/docs/08-03-register-manage-services.md]: That's exactly what I meant :)

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 21]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
     ```
     curl https://gateway.kyma.local:{NODE_PORT}/{APP_NAME}/v1/metadata/services/{YOUR_SERVICE_ID} --cert {CERT_FILE_NAME}.crt --key {KEY_FILE_NAME}.key -k
     ```
+
+
+## Register API with a specification URL
+
+The Application Registry allows you to pass API specifications in a form of specification URLs.
+
+To register API with specification URL replace `api.spec` with `api.specificationUrl`.
```
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]:
```suggestion
To register API with specification URL, replace `api.spec` with `api.specificationUrl`. 
```

File: docs/application-connector/docs/08-03-register-manage-services.md [hunk 22]
```diff
@@ -128,3 +128,46 @@ A successful response returns the ID of the registered service:
...
+
+
+## Register API with a specification URL
+
+The Application Registry allows you to pass API specifications in a form of specification URLs.
+
+To register API with specification URL replace `api.spec` with `api.specificationUrl`. 
+
+>**NOTE** If both api.spec and api.specificationUrl are provided, api.spec will be used due to its higher priority.
+
```
Comment (klaudiagrz) [inline docs/application-connector/docs/08-03-register-manage-services.md]:
```suggestion
See the example of the API part of the request body with specification URL:
```
